chrome/src/background.ts

```diff
@@ -61,6 +61,50 @@ chrome.webNavigation.onBeforeNavigate.addListener(
   },
 )
 
+chrome.tabs.onActivated.addListener(async (activeInfo) => {
+  const tab = await chrome.tabs.query({
+    active: true,
+    lastFocusedWindow: true,
+  })
+
+  const urlString = tab[0].url
+
+  if (!urlString) {
+    return
+  }
+
+  const url = new URL(urlString)
+
+  if (
+    url.host.endsWith('toddle.dev') === false &&
+    url.host.endsWith('nordcraft.com') === false
+  ) {
+    return false
+  }
+
+  // Get the cookies for the .nordcraft.site domain
+  const frames = await chrome.webNavigation.getAllFrames({
+    tabId: activeInfo.tabId,
+  })
+
+  const domain = frames
+    ?.map((frame) => {
+      const url = new URL(frame.url)
+      if (url.host?.endsWith('.toddle.site')) {
+        return url.host
+      }
+    })
+    .filter((v) => v)[0]
+  if (!domain) {
+    return
+  }
+  const cookies = await chrome.cookies.getAll({ domain })
+
+  if (cookies.length > 0) {
+    await updateSessionRules({ domainCookies: cookies, RULE_ID })
+  }
```

Comment:
It seems some of this code is the same as in the `onBeforeNavigate` listener above. Perhaps you could put the shared code in a function? The code that sends information about the cookies to the editor in the `onBeforeNavigate` listener might also be relevant for when a tab receives focus?